Corporate Portal configuration (#1823)
ezsystems/developer-documentation: docs/customer_management/cp_configuration.md

---
description: Configure Customer Portal to fit the needs of your business.
edition: experience
---

# Customer Portal configuration

You can overwrite the default configuration of the Customer Portal
to fit its capabilities to the unique needs of your business.

## `corporate` SiteAccess

The predefined `corporate` SiteAccess in `corporate_group`
(configured in `config/packages/ibexa.yaml`) serves the Customer Portal.
If you need a multisite setup with multiple Customer Portals,
add any additional SiteAccesses to `corporate_group`.

## Customer identifier

> `ibexa_default_settings.yaml` contains setting that indicate which Content Types are treated like customers:

Reviewer: This is not true, that settings just say what content types should be treated like `Users` in terms for example usage in `UserService`.
There is no way to assign which CT should be treated like `cutomer`, as there is no specific behavior for that.

```yaml
ibexa:
    system:
        default:
            user_content_type_identifier: ['user', 'customer']

```

You can override these settings if you have other Content Types
that should be treated as customers in the Back Office and Customer Portal.
When viewing such Content in the Back Office and Customer Portal, you will be able to see e.g. the assigned Policies.

## Roles and Policies

You can easily add custom roles to your installation
by adding them to the configuration file `vendor/ibexa/corporate-account/src/bundle/Resources/config/default_settings.yaml`:

```yaml
parameters:
    ibexa.site_access.config.default.corporate_accounts.roles:
        admin: Company Admin
        buyer: Company Buyer
        custom_role: Company Assistant
```

## Content Types names

> To change names of default Content Types, you need to define them in the configuration

Reviewer: That doesnt make sense, name changing is done in UI, what you could do is assign what content types should be used to describe `Company` and `Member` in Domain layer.
But those configs are done under `ibexa_corporate_account` key, like shown in `\Ibexa\Bundle\CorporateAccount\DependencyInjection\Configuration`.

So if you already have `company` CT in your system, and you dont want to change that identifier, then you should configure CA in following way.
```
ibexa_corporate_account:
    content_type_mappings:
        company: your_ct_identifier
```

Be aware tho, that you have to adjust migration files by yourself if you are in that position.

---
description: Configure Customer Portal to fit the needs of your business.
edition: experience
---

# Customer Portal configuration

You can overwrite the default configuration of the Customer Portal
to fit its capabilities to the unique needs of your business.

## `corporate` SiteAccess

The predefined `corporate` SiteAccess in `corporate_group`
(configured in `config/packages/ibexa.yaml`) serves the Customer Portal.
If you need a multisite setup with multiple Customer Portals,
add any additional SiteAccesses to `corporate_group`.

## Customer identifier

> `ibexa_default_settings.yaml` contains setting that indicate what Content Types should be treated like Users in terms for example usage in `UserService`:

Reviewer: Not sure however, how this relates to customer portal, and it is already mentioned in https://doc.ibexa.co/en/latest/administration/configuration/repository_configuration/#user-identifiers

> `ibexa_default_settings.yaml` contains setting that indicate what Content Types should be treated like Users in terms for example usage in `UserService`:

Reviewer 2: ```suggestion
`ibexa_default_settings.yaml` contains a setting that indicates what Content Types should be treated like Users in terms of, for example, usage in `UserService`:
```

```yaml
ibexa:
    system:
        default:
            user_content_type_identifier: ['user', 'customer']

```

## Roles and Policies

> You can easily add custom roles to your installation

Reviewer 2: ```suggestion
You can add custom roles to your installation
```
"Never say simple" ;)

by adding them to the configuration file `vendor/ibexa/corporate-account/src/bundle/Resources/config/default_settings.yaml`:

```yaml
parameters:
    ibexa.site_access.config.default.corporate_accounts.roles:
        admin: Company Admin
        buyer: Company Buyer
        custom_role: Company Assistant
```

> ## Content Types names

Reviewer 2: ```suggestion
## Content Type names
```

You can change names of default Content Types by assigning what

> Content Types should be used to describe `Company` and `Member` in domain layer.

Reviewer 2: ```suggestion
Content Types should be used to describe `Company` and `Member` in the Back Office.
```
maybe, if that's true?

> Proceed only if you already have `Company` Content Type in your system, and you don't want to change its identifier.

Reviewer 2: ```suggestion
Proceed only if you already have a `Company` Content Type in your system, and you don't want to change its identifier.
```

> Configuration for Content Type names is placed under `ibexa_corporate_account` key,

Reviewer 2: ```suggestion
Configuration for Content Type names is placed under the `ibexa_corporate_account` key,
```

like shown in `Ibexa\Bundle\CorporateAccount\DependencyInjection\Configuration`.

> To change Content Types names, adjust corporate account configuration in the following way:

Reviewer 2: ```suggestion
To change Content Type names, adjust corporate account configuration in the following way:
```

```yaml
ibexa_corporate_account:
    content_type_mappings:
        company: your_ct_identifier
```

!!! caution "Migration"

    If you decide to change deafult names of Content Types, during migration you have to adjust files accordingly.

## Registration

You can define what fields are required in the Customer Portal registration form 
and what Content Type and what identifier customers get after registering.
To do so, go to `vendor/ibexa/corporate-account/src/bundle/Resources/config/default_settings.yaml`.

```yaml
parameters:
    ibexa.site_access.config.corporate_group.user_content_type_identifier: [ 'member' ]
    ibexa.site_access.config.corporate_group.user_registration.user_type_identifier: 'member'
    ibexa.site_access.config.corporate_group.user_registration.form.allowed_field_definitions_identifiers:
        - first_name
        - last_name
        - phone_number
        - user
```

## Address

With the Address Field Type, you can customize address Fields and configure them per country.

> To lern more see [Address Field Type documentation](../content_management/field_types/field_type_reference/addressfield.md).

Reviewer 2: ```suggestion
To learn more, see [Address Field Type documentation](addressfield.md).
```

## Templates

> You can also define new templates for among others: invitation email,

Reviewer 2: ```suggestion
You can also define new templates for, among others: invitation email,
```
